authconfig/oauth.ts

```diff
@@ -0,0 +1,42 @@
+const User = require("../models/Model");
+
+const passport = require("passport");
+const GoogleStrategy = require("passport-google-oauth").OAuth2Strategy;
+
+passport.use(
+  new GoogleStrategy(
+    {
+      clientID: process.env.GOOGLE_ID,
+      clientSecret: process.env.GOOGLE_SECRET,
+      callbackURL: "http://127.0.0.1:3000/auth/google/redirect",
+    },
+    (
+      accessToken: any,
+      refreshToken: any,
+      profile: { id: any; displayName: any },
+      done: (arg0: any, arg1: any) => any
+    ) => {
+      User.findOrCreate({ googleId: profile.id }, function(
+        err: any,
+        user: { save: (arg0: (err: any) => any) => void }
+      ) {
+        if (err) {
+          return done(err, user);
+        }
+        if (!user) {
+          user = new User({
```

Comment:
I think this may be an issue you're having. callind new User({}) would work for mongodb. we need to set user to a variable and just add the required fields to the object